packages/rollup-core/src/app/data/consumers/canonical-chain-batch-submitter.ts

```diff
@@ -246,7 +246,11 @@ export class CanonicalChainBatchSubmitter extends ScheduledTask {
       const gasLimit: string = tx.gasLimit
         ? tx.gasLimit.toString('hex', 64)
         : '00'.repeat(32)
-      const signature: string = remove0x(tx.signature)
+      let signature: string = remove0x(tx.signature)
+      signature =
+        signature.length % 2 === 0
+          ? signature
+          : `${signature.substring(0, 130)}0${signature.substring(130, 133)}`
```

Comment:
Looks about right! Though I must say `v` calculations are incredibly tricky

Reply:
Updated because it's 64 bytes + 2 bytes of V, so it should be 0-128 + 0 + 3 chars of V

Reply:
Crap sorry! Shoulda caught that!

Reply:
But anyway that sounds right